kernel/userspace.c

```diff
@@ -207,6 +207,30 @@ void _dump_object_error(int retval, void *obj, struct _k_object *ko,
 	}
 }
 
+s32_t _impl_k_object_access_check(void *object, struct k_thread *thread)
+{
+	int index;
+	s32_t permission = -EPERM;
```

Comment:
please be consistent in your return values!
-EPERM is a 'true" value when considered as a boolean

Reply:
if the function prototype is a signed integer why would anyone take it as a Boolean? I maybe not clear on this, but if I am returning as "0" that means a success. 
This implementation is similar to that of k_thread_cancel.

Reply:
because your documentation says "+ * This API will return true if the thread has the permission for
 + * the kernel object."!

Instead you are returning boolean FALSE if the thread has permissions

This has to be consistent

Reply:
oh ya, found that problem. Thanks Andrew, fixed it.